src/main/java/ttt/Board.java

```diff
@@ -67,9 +95,9 @@ private boolean checkForWinIn(PlayerSymbol[][] rows) {
     }
 
     private boolean hasWinningColumn() {
-        PlayerSymbol[] leftColumn = new PlayerSymbol[]{grid[0], grid[3], grid[6]};
-        PlayerSymbol[] middleColumn = new PlayerSymbol[]{grid[1], grid[4], grid[7]};
-        PlayerSymbol[] rightColumn = new PlayerSymbol[]{grid[2], grid[5], grid[8]};
+        PlayerSymbol[] leftColumn = new PlayerSymbol[]{grid[0].getSymbol(), grid[3].getSymbol(), grid[6].getSymbol()};
+        PlayerSymbol[] middleColumn = new PlayerSymbol[]{grid[1].getSymbol(), grid[4].getSymbol(), grid[7].getSymbol()};
+        PlayerSymbol[] rightColumn = new PlayerSymbol[]{grid[2].getSymbol(), grid[5].getSymbol(), grid[8].getSymbol()};
 
         PlayerSymbol[][] columns = new PlayerSymbol[][]{leftColumn, middleColumn, rightColumn};
```

Comment:
There is a lot of complexity and responsibilities in this method.

It calculates the columns to then do it's job of passing the columns to the `checkForWinningIn(columns)` method.

A nicer way to do this would be to extract the column generation to methods:

```
private PlayerSymbol[] leftColumn() {
    return new PlayerSymbol[]{grid[0].getSymbol(), grid[3].getSymbol(), grid[6].getSymbol()};
}
```

Doing this will give you the following code for the `hasWinningColumn ` method:
private boolean hasWinningColumn() {
    PlayerSymbol[][] columns = new PlayerSymbol[][]{leftColumn(), middleColumn(), rightColumn()};
    return checkForWinIn(columns);
}